app/services/candidate_submission.rb

```diff
@@ -0,0 +1,71 @@
+class CandidateSubmission
+  include Candidate
+
+  def initialize(session, step_name)
+    @session = session
+    @step_name = step_name
+  end
+
+  def call
+    sign_up_candidate(select_candidate)
+  end
+
+private
+
+  def sign_up_candidate(body)
+    ApiClient.sign_up_teacher_training_adviser_candidate(body)
+  end
+
+  def select_candidate
+    case @step_name
+    when "accept_privacy_policy"
```

Comment:
Determining the type of candidate based on the policy they have accepted feels fickle to me - can we not set an explicit variable in the session with the type of candidate?